opensearch-project/OpenSearch-Dashboards
# [MD] Change cluster selector component name to data source selector

### Description
This change renames clusterselector to datasource selector 

### Issues Resolved
Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6038

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/e7767763-1db2-442f-a9ac-0f3b961c64ec


## Testing the changes
The following steps were performed in the recording above:
1. enable data source plugin
2. add sample data should render data source selector correctly and can select data source and add sample data, and sample data should render correctly
3. in dev tools, select a data source and perform query should render results correctly
4. go to query workbench, should render data source selector correctly and can select data source and request to yield indices
5. go to alert plugin, should render data source selector correctly and can select data source and request to yield monitors from data source

### Check List
